java-spiffe-core/src/main/java/io/spiffe/internal/JwtSignatureAlgorithm.java

```diff
@@ -128,7 +124,7 @@ public static Algorithm parse(final String s) {
         } else if (s.equals(PS512.getName())) {
             algorithm = PS512;
         } else {
-            algorithm = OTHER;
+            throw new IllegalArgumentException("Unsupported JWT token algorithm: " + s);
```

Comment:
nit: `JWT token` is a little redundant, I think just removing `token` from the message should be fine
```suggestion
            throw new IllegalArgumentException("Unsupported JWT algorithm: " + s);
```